shipitscript/shipitscript/pushlog_scan.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from enum import Enum, auto, unique
+
+import requests
+
+log = logging.getLogger(__name__)
+
+URL = 'https://hg.mozilla.org/{branch}/json-pushes'
+
+
+@unique
+class Importance(Enum):
+    MAYBE = auto()
+    IMPORTANT = auto()
+    UNIMPORTANT = auto()
+
+
+_push_checks = []
+
+
+def push_check(func):
+    _push_checks.append(func)
+    return func
+
+
+def get_shippable_revision_build(branch, last_shipped_rev, cron_rev):
+    """This method queries the hg.mozilla.org server and retrieves all the commits
+    that have happened between `last_shipped_rev` and the `cron_rev`. Sorted
+    decreasingly over time, it iterates through them and attempts to retrieve the
+    first one that is important. If none found, it'll return None and the main
+    function will make the task go green.
+    """
+    # TODO use redo here to automatically retry
+    resp = requests.get(
+        URL.format(branch=branch),
+        params={'full': '1', 'fromchange': last_shipped_rev, 'tochange': cron_rev},
+    )
+    pushlog = resp.json()
+    reversed_pushes = [pushlog[k] for k in sorted(pushlog.keys(), reverse=True)]
+    for push in reversed_pushes:
+        push_importance = is_push_important(push)
+        if push_importance == Importance.IMPORTANT:
+            # Tell caller that we want to build.
+            return push['changesets'][-1]['node']
```

Comment:
so `node` will get revision?

Also, we are still ignoring the case where we are returning a revision that includes DONTBUILD?

Reply:
I'm okay with this being a follow up. As per slack.

Reply:
Added another decorator, please let me know if it looks good.